src/Microsoft.AspNetCore.Routing/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -261,6 +266,19 @@ private Endpoint CreateRejectionEndpoint(IEnumerable<string> httpMethods)
                 Http405EndpointDisplayName);
         }
 
+        public static bool ContainsHttpMethod(List<string> httpMethods, string httpMethod)
+        {
+            for (int i = 0; i < httpMethods.Count; i++)
```

Comment:
Wouldn't it be better to either use a foreach or introduce a local for the `httpMethods.Count`?
#askben